core/integration/steps/the_markets.go

```diff
@@ -84,10 +84,16 @@ func TheMarkets(
 		return nil, err
 	}
 
-	if err := submitMarkets(markets, executionEngine, now); err != nil {
-		return nil, err
+	for i, row := range rows {
+		if err := executionEngine.SubmitMarket(context.Background(), &markets[i], "proposerID", now); err != nil {
+			return nil, fmt.Errorf("couldn't submit market(%s): %v", markets[i].ID, err)
+		}
+		if !row.HasColumn("is passed") || row.Bool("is passed") {
+			if err := executionEngine.StartOpeningAuction(context.Background(), markets[i].ID); err != nil {
+				return nil, fmt.Errorf("could not start opening auction for market %s: %v", markets[i].ID, err)
+			}
```

Comment:
Why take this out of the function?

Reply:
The `is passed` logic is added so we can get a market to `STATUS_PROPOSED` without going straight to `STATUS_PASSED`.

Moving it out the function was purely so not to have to pass the `rows` slice into the function and the fact it isn't used elsewhere. Do you think it's better left in?